File: backend/src/main/java/com/lacrisalide/model/Event.java
package com.lacrisalide.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "events")
public class Event {
 @Id
 @GeneratedValue(strategy = GenerationType.IDENTITY)
 private Long id;

 @Column(nullable = false, length = 180)
 private String titolo;

 @Column(nullable = false, length = 3000)
 private String descrizione;

 @Column(nullable = false)
 private LocalDateTime data;

 private LocalDateTime dataFine;

 @Column(nullable = false, length = 240)
 private String luogo;

 @Column(nullable = false)
 private Integer maxPartecipanti;

 @Column(columnDefinition = "LONGTEXT")
 private String volantino;

 @Column(nullable = false, updatable = false)
 private LocalDateTime createdAt;

 @PrePersist
 protected void onCreate() {
  createdAt = LocalDateTime.now();
 }
}

File: backend/Models/Event.cs
namespace SoccerDreamFermana.Backend.Models;

public sealed class Event
{
    public long Id { get; set; }
    public string Titolo { get; set; } = "";
    public string? Tipo { get; set; }
    public string Descrizione { get; set; } = "";
    public DateTime Data { get; set; }
    public DateTime? DataFine { get; set; }
    public string Luogo { get; set; } = "";
    public int MaxPartecipanti { get; set; }
    public bool UnlimitedCapacity { get; set; }
    public string? Volantino { get; set; }
    public DateTime CreatedAt { get; set; }
    public List<Booking> Bookings { get; set; } = [];
}
